satpy/readers/tropomi_l2.py

```diff
@@ -158,6 +163,30 @@ def _rename_dims(self, data_arr):
             dims_dict['scanline'] = 'y'
         return data_arr.rename(dims_dict)
 
+    def prepare_geo(self, bounds_data):
+        # lat/lon bounds are ordered in the following way:
+        # 3----2
+        # |    |
+        # 0----1
+        # Extend longitudes and latitudes with one element to support "pcolormesh"
+        dest_shape = (bounds_data.shape[0]+1, bounds_data.shape[1]+1)
+        dest = np.zeros(dest_shape, dtype=np.float64)
```

Comment:
I haven't checked in this reader, but isn't `bounds_data` a `xr.DataArray` ? in this case, maybe we could preserve the lazyness by assembling the bounds using `np.hstack` and `np.vstack` for example instead of direct assigment ?

Reply:
@mraspaud 
Do you mean that I need to create two 1D arrays myself, stack them to `bounds_data` and then replace the values?

Reply:
I was thinking something like (untested):
```python
bottom = np.hstack([bounds_data[:, :, 0],  bounds_data[:, -1, 1]])
top = np.hstack([bounds_data[-1, :, 3], bounds_data[-1, -1, 2]])
dest = np.vstack([top, bottom])
```

Reply:
@mraspaud Good idea!

But, `bounds_data[:, :, 0]` is 2D array and `bounds_data[:, -1, 1]` is 1D array. It raised this error:
```
ValueError: all the input arrays must have same number of dimensions, but the array at index 0 has 2 dimension(s) and the array at index 1 has 1 dimension(s)
```

So, I made this trick to make the 2D array:
```
bottom = np.hstack([bounds_data[:, :, 0],  bounds_data[:, -1:, 1]])
```